File: CSCI 463/Print-Conform-main/scripts/python/connect_db.py
# PrintConform Connect DB - Python
# Version: 1.0
# Purpose: Used to connect to the PrintConform database. 
# Instructions: File should be kept out of root, and only be accessable from PHP user (CHMOD ???).
# In the python program that needs to call the MySQL database: "import connect-db" and close the connection via conn.close().
# Author: Brandon Veen
# Creation Date: 5-2-2024
# Modification History:
### 5-2-2024 - First version completed.

import mysql.connector
#from sshtunnel import SSHTunnelForwarder

def connect_db():
    #Define config
    config = {
        'user': 'printconform',
        'password': 'FUtBt!4Oqs',
        'host': 'mysql.printconform.com',
        'database': 'printconform',
        'raise_on_warnings': True
    }
    #Open Connection to MySQL
    try:
        conn = mysql.connector.connect(**config)
    except mysql.connector.Error as err:
        print(err)
        conn = -1
    
    #Return connection to calling function.
    return(conn)

File: CSCI 463/Print-Conform-main/scripts/php/connect_db.php
<?php 
    /* Print Conform Connect DB
        * Version: 1.0
        * Purpose: For inclusion in python scripts, stores database connection information.
        * Author: Brandon Veen
        * Creation Date: 05-04-2024
    */
    // MySQL Connection Information
    $servername = "mysql.printconform.com";
    $username = "printconform";
    $password = "FUtBt!4Oqs";
    $schema = "printconform";
    $conn = new mysqli($servername, $username, $password, $schema);
    unset($servername);
    unset($username);
    unset($password);
    unset($schema);
?>
